### ISSUE TITLE
Custom shapes for nodes

### ISSUE THREAD

**Issue by vshamgin** [Netflix/vizceral#92]
It would be nice to have have a mechanism to render custom node shapes (instead of circles) based on data, passed in json like:
```
      "nodes": [
        {
          "name": "blah",
          "displayName" : "Blah123",
          "node_type": "my_custom_node"
        },..
```

These should be "shapes", since due to the current implementation (canvas, 2D) of Vizceral and potential performance issues, it is expensive to use 3D objects (imported .3ds). Also, using textures on 2D planes is not an option - they look bad when a node zooms in (100% + ) when it been double-clicked.

Something like this:
https://www.dropbox.com/sh/bvfwaw3p2r7j2by/AACAkrxXZK76982lz4s-JMyra?dl=0

**Comment by vshamgin** [comment]
The PR is here:
https://github.com/Netflix/vizceral/pull/93
Thanks!

**Comment by prabhurangan** [comment]
@vshamgin Is this PR merged??

**Comment by vshamgin** [comment]
@prabhurangan still pending :)
https://github.com/Netflix/vizceral/pull/93

**Comment by jrsquared** [comment]
As soon as the copyright headers are removed, I can get this merged

**Comment by vshamgin** [comment]
@jrsquared Removed copyright headers from the new files:
https://github.com/Netflix/vizceral/pull/93/commits/aa165f66b15fa91d86a24dbe6dd647eefa846b9e

Thanks!
~Vlad

**Comment by prabhurangan** [comment]
@jrsquared Once it is merged, i can get the changes by npm update?

**Comment by jrsquared** [comment]
Implemented in #93

**Comment by prabhurangan** [comment]
@jrsquared will it be available in Vizceral 4.5.6?

**Comment by prabhurangan** [comment]
@jrsquared sorry ignore my previous comment, got the update in 4.6.0.  Thank you

**Comment by anduingaiden** [comment]
Hi there, I tried to use the CustomShapes but with no success. Here it goes a part of my sample_data.json (which comes with vizceral-example):
`{
  "renderer": "global",
  "name": "edge",
  "nodes": [
    {
      "renderer": "region",
      "name": "INTERNET",
      "node_type": "service",
      "displayName": "INTERNET",
      "nodes": [],
      "metadata": {},
      "class": "normal",
      "connections": []
    },
    {
      "renderer": "region",
      "name": "farfetch_com_azure_we",
      "displayName": "FF_COM_AZURE_WE",
      "updated": 1477690448572,
      "nodes": [
        {
          "name": "viraemic",
          "metadata": {
            "streaming": 1
          },
          "nodes": [
            {
              "name": "tetrabrach",
              "metrics": {
                "normal": 41515.944,
                "danger": 66.144
              }
            },
            {
              "name": "colloidal",
              "metrics": {
                "danger": 0.166,
                "normal": 126.34000000000002
              }
            },
            {
              "name": "wardrobers"
            },
            {
              "name": "yplast"
            },
            {
              "name": "benet",
              "metrics": {
                "danger": 0.22400000000000003,
                "normal": 130.17600000000002
              }
            },
            {
              "name": "imping",
              "metrics": {
                "danger": 0.22000000000000003,
                "normal": 130.19000000000003
              }
            },
            {
              "name": "virility",
              "metrics": {
                "danger": 0.18600000000000003,
                "normal": 130.158
              }
            },
            {
              "name": "eng",
              "metrics": {
                "danger": 0.244,
                "normal": 129.494
              }
            },
            {
              "name": "use",
              "metrics": {
                "danger": 0.2,
                "normal": 126.536
              }
            },
            {
              "name": "racegoings",
              "metrics": {
                "danger": 0.06999999999999999,
                "normal": 43.19200000000001
              }
            }
          ],
          "renderer": "focusedChild",
          "node_type": "storage"
        },..........continues blablabla`

What am I doing wrong?

**Comment by thebrandonhoward** [comment]
Was the Custom Shapes functionality removed?

**Comment by vshamgin** [comment]
@jrsquared 
I will be able to take a look soon

**Comment by thebrandonhoward** [comment]
Great. I’ll fork and contribute also.
